src/devices/Scd4x/Scd4x.cs

```diff
@@ -0,0 +1,240 @@
+        }
+
+        /// <summary>
+        /// Begins calibrating the sensor to operate at a specific barometric pressure.
+        /// Doing so will make measurements more accurate.
+        /// </summary>
+        /// <param name="pressure">The pressure to use when calibrating the sensor.</param>
+        /// <returns>
+        /// The time that the caller must wait before performing any further operations on the device.
+        /// </returns>
+        public TimeSpan BeginSetPressureCalibration(Pressure pressure)
+        {
+            Span<byte> buffer = stackalloc byte[5];
+
+            BinaryPrimitives.WriteUInt16BigEndian(buffer, 0xE000);
+            Sensirion.WriteUInt16BigEndianAndCRC8(buffer.Slice(2), (ushort)(Math.Max(0.0, Math.Min(pressure.Pascals, 1.0)) * (1.0 / 100.0)));
+
+            _device.Write(buffer);
+            return TimeSpan.FromTicks(TimeSpan.TicksPerMillisecond);
+        }
+
+        /// <summary>
+        /// <para>
+        /// Instructs the sensor to start performing periodic measurements.
+        /// </para>
+        ///
+        /// <para>
+        /// Every period, the length of which is available in <see cref="MeasurementPeriod"/>, <see cref="CheckDataReady"/> can be called to see if a measurement is available,
+        /// and the measurement can then be read via <see cref="ReadPeriodicMeasurement"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// <see cref="StopPeriodicMeasurements"/> must be called to stop periodic measurements.
+        /// </para>
+        /// </summary>
+        public void StartPeriodicMeasurements() =>
+            _device.Write(StartPeriodicMeasurementBytes);
+
+        /// <summary>
+        /// <para>
+        /// Checks if a periodic measurement is available.
+        /// Once available, the measurement can be read via <see cref="ReadPeriodicMeasurement"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// <see cref="StartPeriodicMeasurements"/> must be called first.
+        /// </para>
+        /// </summary>
+        /// <returns>If a measurement is available, <see langword="true"/>. Otherwise, <see langword="false"/>.</returns>
+        public bool CheckDataReady()
+        {
+            TimeSpan delay = BeginCheckDataReady();
+            Thread.Sleep(delay);
+            return EndCheckDataReady();
+        }
+
+        /// <summary>
+        /// <para>
+        /// Begins checking if a periodic measurement is available.
+        /// Once available, the measurement can be read via <see cref="ReadPeriodicMeasurement"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// <see cref="StartPeriodicMeasurements"/> must be called first.
+        /// </para>
+        /// </summary>
+        /// <returns>
+        /// The time that the caller must wait before calling <see cref="EndCheckDataReady"/> to retreive the current data ready status.
+        /// </returns>
+        public TimeSpan BeginCheckDataReady()
```

Comment:
What is the benefit of having this method be public? Same goes for EndCheckDataReady method. Seems to me these could be private as they get called by CheckDataReady method.